packages/ra-ui-materialui/src/button/BulkDeleteWithConfirmButton.js

```diff
@@ -106,7 +106,7 @@ class BulkDeleteWithConfirmButton extends Component {
                     translateOptions={{
                         smart_count: selectedIds.length,
                         name: inflection.humanize(
-                            inflection.singularize(resource),
+                            inflection.inflect(resource, selectedIds.length),
```

Comment:
I think we should do even better. Many of users actually provide translations for their resources names. We should use them

Reply:
See #3040

Reply:
Good idea, but this should be done across all messages. Let's merge @Luwangel's PR and do that in another PR.